.github/push-images.sh

```diff
@@ -10,3 +10,39 @@ do
     echo "Pushing ${IMG}"
     docker push ${IMG}
 done
+
+manifest_push () {
+    local image=$1
+    local tag=$2
+
+    docker manifest create quay.io/quarkus/${image}:${tag} \
+        --amend quay.io/quarkus/${image}:${tag}-amd64 \
+        --amend quay.io/quarkus/${image}:${tag}-arm64
+    docker manifest push quay.io/quarkus/${image}:${tag}
+}
+
+# Manually combine manifests for multi-arch native builder images
+if [[ $(docker images | grep quay.io/quarkus/ | grep "ubi-quarkus-native-image") ]]; then
+    manifest_push "ubi-quarkus-native-image" "22.0.0-java17"
+    manifest_push "ubi-quarkus-native-image" "22.0.0-java11"
+    manifest_push "ubi-quarkus-native-image" "22.0-java17"
+    manifest_push "ubi-quarkus-native-image" "22.0-java11"
+
+    manifest_push "ubi-quarkus-native-image" "22.1.0-java17"
+    manifest_push "ubi-quarkus-native-image" "22.1.0-java11"
+    manifest_push "ubi-quarkus-native-image" "22.1-java17"
+    manifest_push "ubi-quarkus-native-image" "22.1-java11"
+fi
+
+# Manually combine manifests for multi-arch mandrel images
+if [[ $(docker images | grep quay.io/quarkus/ | grep "ubi-quarkus-mandrel") ]]; then
+    manifest_push "ubi-quarkus-mandrel" "22.0.0.2.Final-java17"
+    manifest_push "ubi-quarkus-mandrel" "22.0.0.2.Final-java11"
+    manifest_push "ubi-quarkus-mandrel" "22.0-java17"
+    manifest_push "ubi-quarkus-mandrel" "22.0-java11"
+
+    manifest_push "ubi-quarkus-mandrel" "22.1.0.0.Final-java17"
+    manifest_push "ubi-quarkus-mandrel" "22.1.0.0.Final-java11"
+    manifest_push "ubi-quarkus-mandrel" "22.1-java17"
+    manifest_push "ubi-quarkus-mandrel" "22.1-java11"
```

Comment:
We should definitely find a better way to figure out the tags, ideally fetching them from https://github.com/quarkusio/quarkus-images/blob/54076338ebb857c26b5c8477b11562355aaa59b3/.github/mandrel-images.yaml#L4

If jbang is the preferred way I am OK with that. After all I never had to touch "push-images" before, so I would prefer the people actually maintaining it to decide what they prefer.

Reply:
When I first discussed this with @cescoffier, I agreed with your statement: let's make this a jbang script and consume the versions in yaml, but I don't think it'll really solve the little hardcoding done in the bash script.

Put it this way: none of the image versions that need to be created (e.g. mandrel `22.1.0.0.Final-java17` or graalvm `22.0.0-java17`) are part of the `versions` that you pass in in the yaml. The versions that you pass in are the ones you build the traditional way (same for build and push) and then you have these extra overarching versions that are only used by push and created and pushed differently (e.g. `docker manifest create...` and `docker manifest push`).

So, I'm currently leaning towards keeping things as they are in the PR. One thing that complicates making wide ranging changes in the push side is that there's no real way to test it until you get the GH action to kick in and see if it works as expected. Note that I actually tested something very similar to this in my own repo's `main` branch and a custom GH action. That's how I ended up pushing multi-arch images to my own fork, see [here](https://quay.io/repository/gzamarre/ubi-quarkus-native-image?tab=tags).

Reply:
It's true as well that you could enhance the yaml to somehow add the multi arch tags (something akin to what happens with `tags`) and then have some kind of way of saying that this multi-arch tag is built out of these other tags. E.g.

```
multi-arch-tags:
  - id: 22.1-java17-amd64
    manifest-versions:
    - 22.1-java17-amd64
    - 22.1-java17-arm64
  - id: 22.1.0-java17-amd64
    manifest-versions:
    - 22.1.0-java17-amd64
    - 22.1.0-java17-arm64
```